### PR TITLE
[SPARK-7156][SQL] add randomSplit to DataFrame.

### PR DESCRIPTION
**Pull request by kaka1992**
SPARK-7156 add randomSplit to DataFrame.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala [hunk 1]
```diff
@@ -967,6 +969,23 @@ class DataFrame private[sql](
...
+   * @param seed random seed
+   *
+   * @return split DataFrames in an array
+   */
+  def randomSplit(weights: Array[Double], seed: Long = Utils.random.nextLong): Array[DataFrame] = {
+    val sum = weights.sum
+    val normalizedCumWeights = weights.map(_ / sum).scanLeft(0.0d)(_ + _)
+    normalizedCumWeights.sliding(2).map { x =>
+      this.sqlContext.createDataFrame(new PartitionwiseSampledRDD[Row, Row](
+        rdd, new BernoulliCellSampler[Row](x(0), x(1)), true, seed), schema)
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala]: this actually breaks the plan -- can we create a logical operator (or generalizes the existing Sample operator) so the returned DataFrame correctly preserves the logical plan?

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala [hunk 2]
```diff
@@ -17,14 +17,13 @@
 
 package org.apache.spark.sql
 
-import scala.language.postfixOps
```
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala]: scala inports should be first

**Comment by rxin** [comment]
Thanks for working on this, @kaka1992. Would be great if we can do it in a way that doesn't break the existing logical plan for data frames.

**Comment by kaka1992** [comment]
Can I add InMemoryRelation upon the base logicalPlan? Then I could create several randomSplit plans with the same data. @rxin I'm not sure if this way would break something.

**Comment by rxin** [comment]
https://github.com/apache/spark/pull/5761 

Somebody else submitted a PR based on your change and my review feedback.

**Comment by rxin** [comment]
@kaka1992 mind closing the pr since https://github.com/apache/spark/pull/5761 subsumes this?

**Comment by kaka1992** [comment]
@rxin No problem. I'll close the pr.

**Pull request closed without merging**